com.unity.render-pipelines.universal/Runtime/2D/Passes/Utility/RendererLighting.cs

```diff
@@ -162,32 +163,72 @@ private static void RenderLightSet(IRenderPass2D pass, RenderingData renderingDa
 
                     ShadowRendering.RenderShadows(pass, renderingData, cmd, layerToRender, light, light.shadowIntensity, renderTexture, renderTexture);
 
+                    // Still used by non-Batched code.
+                    cmd.SetGlobalColor(k_LightColorID, light.intensity * light.color);
+
                     if (light.lightType == Light2D.LightType.Sprite && light.lightCookieSprite != null && light.lightCookieSprite.texture != null)
                         cmd.SetGlobalTexture(k_CookieTexID, light.lightCookieSprite.texture);
 
-                    cmd.SetGlobalFloat(k_FalloffIntensityID, light.falloffIntensity);
-                    cmd.SetGlobalColor(k_LightColorID, light.intensity * light.color);
-                    cmd.SetGlobalFloat(k_VolumeOpacityID, light.volumeOpacity);
-
                     if (light.useNormalMap || light.lightType == Light2D.LightType.Point)
                         SetPointLightShaderGlobals(pass, cmd, light);
 
-                    // Light code could be combined...
-                    if (light.lightType == Light2D.LightType.Parametric || light.lightType == Light2D.LightType.Freeform || light.lightType == Light2D.LightType.Sprite)
+                    // Batch if possible.
+                    bool skipDraw = false;
+                    if (batchable)
                     {
-                        cmd.DrawMesh(lightMesh, light.transform.localToWorldMatrix, lightMaterial);
+                        if (light.lightType == Light2D.LightType.Parametric ||
+                            light.lightType == Light2D.LightType.Freeform)
+                        {
+                            if (light.shadowIntensity == 0)
+                            {
+                                if (Light2DBatch.sActiveMaterial == null || lightMaterial != Light2DBatch.sActiveMaterial)
+                                {
+                                    // If this is not the same material, end any previous valid batch.
+                                    if (Light2DBatch.sActiveMaterial)
+                                        Light2DBatch.EndBatch(cmd);
+                                    Light2DBatch.StartBatch(lightMaterial);
+                                }
+
+                                Light2DBatch.AddMesh(lightMesh, light.transform, light.lightMeshHash);
+                                skipDraw = true;
+                            }
+                        }
                     }
-                    else if (light.lightType == Light2D.LightType.Point)
+
+                    if (skipDraw)
```

Comment:
if we are to batch, let's rework this loop to a simpler one... this loop should figure out where batching ends and then call a `DrawBatch(startIndex, count)` keeps the code cleaner

Reply:
Agree, will fix.

Reply:
Simplified and fixed in 298c27472e100a9d7d5ace7fdb4825ae0b6ea6c5